From: Suren Baghdasaryan <surenb@google.com>
Subject: arch/um: fix forward declaration for vmalloc
Date: Tue, 26 Mar 2024 00:37:50 -0700

Patch [1] replaced vmalloc() function with a new definition but it did not
adjust the forward declaration used in UML architecture.  Change it to act
as before.

Note that this prevents the vmalloc() allocations in __wrap_malloc() from
being accounted.  If accounting here is critical, we will have to remove
this forward declaration and include vmalloc.h, however that would pull in
more dependencies and would require introducing more architecture-specific
headers, like asm/bug.h, asm/rwonce.h, etc.  This is likely the reason why
this forward declaration was introduced in the first place.

[1] https://lore.kernel.org/all/20240321163705.3067592-31-surenb@google.com/

Link: https://lkml.kernel.org/r/20240326073750.726636-1-surenb@google.com
Fixes: 576477564ede ("mm: vmalloc: enable memory allocation profiling")
Signed-off-by: Suren Baghdasaryan <surenb@google.com>
Reported-by: SeongJae Park <sj@kernel.org>
Closes: https://lore.kernel.org/all/20240323180506.195396-1-sj@kernel.org/
Tested-by: SeongJae Park <sj@kernel.org>
Reviewed-by: Johannes Berg <johannes@sipsolutions.net
Acked-by: Richard Weinberger <richard@nod.at>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/um/include/shared/um_malloc.h |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/arch/um/include/shared/um_malloc.h~mm-vmalloc-enable-memory-allocation-profiling-fix
+++ a/arch/um/include/shared/um_malloc.h
@@ -11,7 +11,8 @@
 extern void *uml_kmalloc(int size, int flags);
 extern void kfree(const void *ptr);
 
-extern void *vmalloc(unsigned long size);
+extern void *vmalloc_noprof(unsigned long size);
+#define vmalloc(...)		vmalloc_noprof(__VA_ARGS__)
 extern void vfree(void *ptr);
 
 #endif /* __UM_MALLOC_H__ */
_
